src/notification.rs

```diff
@@ -233,3 +236,47 @@ impl Notification for PublishDiagnostics {
     type Params = PublishDiagnosticsParams;
     const METHOD: &'static str = "textDocument/publishDiagnostics";
 }
+
+
+#[cfg(test)]
+mod test {
+    use std::mem;
+    use super::*;
+
+    fn fake_call<N>(_params: N::Params)
+        where N: Notification,
+              N::Params: serde::Serialize,
+    { }
+
+    macro_rules! check_macro {
+        ($name:tt) => {
+            // check whethe the macro name matches the method
+            assert_eq!(<lsp_notification!($name) as Notification>::METHOD, $name);
+            // test whether type checking passes for each component
+            if false {
+                // don't want to actually implement/run this
+                let params: <lsp_notification!($name) as Notification>::Params = unsafe { mem::uninitialized() };
+                fake_call::<lsp_notification!($name)>(params);
```

Comment:
I think there is no need to create an uninitialized value here. Calling `fake_call::<lsp_notification!($name)>()` without any parameters should be enough to validate everything.